packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -108,6 +108,30 @@ class _Compiler {
         .createTempSync('output_dill');
     final File outputDill = outputDillDirectory.childFile('output.dill');
 
+    bool suppressOutput = false;
+    void reportCompilerMessage(String message) {
+      if (suppressOutput)
+        return;
+
+      if (message.startsWith('compiler message: Error: Could not resolve the package \'test\'')) {
```

Comment:
What is the reason for this logic and supporting infrastructure(like suppressOutput, reportCompilerMessage, compilerMessageConsumer)? Why do we handle this error specially and suppress the rest of compiler output?

Reply:
There is a test that tests that a nice message is produced when somebody tries to do `flutter test` on a wrong file (search for 'Failed to load test harness' in the whole file to find that place).

The suppression I added just to make output. We can definitely improve this later even further.

Reply:
> The suppression I added just to make output.

Not sure I understand. It looks like the only place we use suppressOutput(set it to true) is here when we encounter this error. With this special handling we won't need to introduce suppressOutput flag, reportCompilerMessage function, compilerMessageConsumer parameter. Am I reading it wrong?